src/app-layer-htp-file.c

```diff
@@ -397,6 +432,38 @@ int HTPFileStoreChunk(HtpState *s, const uint8_t *data, uint32_t data_len,
     SCReturnInt(retval);
 }
 
+static void HTPFileCloseHandleRange(FileContainer *files, const uint8_t flags,
+        HttpRangeContainerBlock *c, const uint8_t *data, uint32_t data_len)
+{
+    if (c->container) {
+        THashDataLock(c->container->hdata);
+        if (c->container->error) {
+            SCLogDebug("range in ERROR state");
+        }
+        if (HttpRangeAppendData(c, data, data_len) < 0) {
+            SCLogDebug("Failed to append data");
+        }
+        File *ranged = HttpRangeClose(c, flags);
+        if (ranged) {
+            /* HtpState owns the constructed file now */
+            FileContainerAdd(files, ranged);
+        }
+        SCLogDebug("c->container->files->tail %p", c->container->files->tail);
+        THashDataUnlock(c->container->hdata);
+    } else {
+        if (c->toskip > 0) {
+            if (HttpRangeProcessSkip(c, data, data_len) < 0) {
+                SCLogDebug("Failed to append data");
+            }
+        }
+        File *ranged = HttpRangeClose(c, flags);
```

Comment:
Is there anything to close if `c->container == NULL` ?

Reply:
Good catch. I had been meaning to revisit, but forgot. No it looks like this can go.